src/transformers/trainer.py

```diff
@@ -813,7 +813,9 @@ def create_optimizer(self):
         if is_sagemaker_mp_enabled():
             self.optimizer = smp.DistributedOptimizer(self.optimizer)
 
-    def create_scheduler(self, num_training_steps: int):
+        return self.optimizer
+
+    def create_scheduler(self, num_training_steps: int, optimizer: torch.optim.Optimizer = None):
         """
         Setup the scheduler. The optimizer of the trainer must have been set up before this method is called.
```

Comment:
```suggestion
        Setup the scheduler. The optimizer of the trainer must have been set up either before this method is called or passed as an argument.
```

need to update the doc too